d2common/d2fileformats/d2tbl/text_dictionary.go

```diff
@@ -156,12 +193,78 @@ func LoadTextDictionary(dictionaryData []byte) (TextDictionary, error) {
 		}
 	}
 
-	hashEntries := make([]*textDictionaryHashEntry, hashTableSize)
+	lookupTable.elementIndex = elementIndex
 
-	err = lookupTable.loadHashEntries(hashEntries, br)
+	lookupTable.hashEntries = make([]*textDictionaryHashEntry, lookupTable.hashTableSize)
+
+	err = lookupTable.loadHashEntries(br)
 	if err != nil {
 		return nil, err
 	}
 
 	return lookupTable, nil
 }
+
+// Marshal encodes text dictionary back to byte slice
+func (td *TextDictionary) Marshal() []byte {
+	// create stream writter
+	sw := d2datautils.CreateStreamWriter()
+
+	sw.PushBytes(td.crcBytes...)
```

Comment:
is the CRC never calculated... ?

Reply:
https://godoc.org/github.com/howeyc/crc16

Reply:
I think, it isn't related with this PR.
I've opened a separated issue for it https://github.com/OpenDiablo2/OpenDiablo2/issues/1043

Reply:
would be a nice unit test